src/com/java/multithreading/BlockingQueue.java

```diff
@@ -0,0 +1,46 @@
+package com.java.multithreading;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class BlockingQueue {
+
+  private Queue<Integer> queue;
```

Comment:
Why just accepting interger?...make it generic, so that it can accept any data type.